machine/topology_test.go

```diff
@@ -81,98 +135,82 @@ func TestTopology(t *testing.T) {
+		t.Errorf("Expected empty sysfs to fail.")
 	}
 }
 
-func TestTopologyCoreId(t *testing.T) {
-	val, _ := getCoreIdFromCpuBus("./testdata", 0)
-	if val != 0 {
-		t.Errorf("Expected core 0, found %d", val)
+func TestTopologyWithNodesWithoutCPU(t *testing.T) {
+	sysFs := &fakesysfs.FakeSysFs{}
+	nodesPaths := []string{
+		"/fakeSysfs/devices/system/node/node0",
+		"/fakeSysfs/devices/system/node/node1",
 	}
+	sysFs.SetNodesPaths(nodesPaths, nil)
 
-	val, _ = getCoreIdFromCpuBus("./testdata", 9999)
-	if val != 8888 {
-		t.Errorf("Expected core 8888, found %d", val)
-	}
-}
+	memTotal := []byte("MemTotal:       32817192 kB")
+	sysFs.SetMemory(memTotal, nil)
 
-func TestTopologyNodeId(t *testing.T) {
-	val, _ := getNodeIdFromCpuBus("./testdata", 0)
-	if val != 0 {
-		t.Errorf("Expected core 0, found %d", val)
+	hugePages := []os.FileInfo{
+		&fakesysfs.FileInfo{EntryName: "hugepages-2048kB"},
+		&fakesysfs.FileInfo{EntryName: "hugepages-1048576kB"},
 	}
+	sysFs.SetHugePages(hugePages, nil)
 
-	val, _ = getNodeIdFromCpuBus("./testdata", 9999)
-	if val != 1234 {
-		t.Errorf("Expected core 1234 , found %d", val)
+	hugePageNr := map[string][]byte{
+		"/fakeSysfs/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages":    []byte("1"),
+		"/fakeSysfs/devices/system/node/node0/hugepages/hugepages-1048576kB/nr_hugepages": []byte("1"),
+		"/fakeSysfs/devices/system/node/node1/hugepages/hugepages-2048kB/nr_hugepages":    []byte("1"),
+		"/fakeSysfs/devices/system/node/node1/hugepages/hugepages-1048576kB/nr_hugepages": []byte("1"),
 	}
-}
+	sysFs.SetHugePagesNr(hugePageNr, nil)
 
-func TestGetHugePagesInfo(t *testing.T) {
-	testPath := "./testdata/hugepages/"
-	expected := []info.HugePagesInfo{
-		{
-			NumPages: 1,
-			PageSize: 1048576,
-		},
-		{
-			NumPages: 2,
-			PageSize: 2048,
-		},
-	}
+	topology, numCores, err := GetTopology(sysFs)
 
-	val, err := GetHugePagesInfo(testPath)
-	if err != nil {
-		t.Errorf("Failed to GetHugePagesInfo() for sample path %s: %v", testPath, err)
-	}
+	assert.Nil(t, err)
+	assert.Equal(t, 0, numCores)
 
-	if !reflect.DeepEqual(expected, val) {
-		t.Errorf("Expected HugePagesInfo %+v, got %+v", expected, val)
-	}
+	topologyJSON, err := json.Marshal(topology)
+	if err != nil {
+		t.Errorf("failed to marsha topology, %v", err)
+	}
+	expectedTopology := []byte(
+		`[
+     {
+      "caches": null,
+      "cores": null,
+      "hugepages": [
+       {
+        "num_pages": 1,
+        "page_size": 2048
+       },
+       {
+        "num_pages": 1,
+        "page_size": 1048576
+       }
+      ],
+      "memory": 33604804608,
+      "node_id": 0
+     },
+     {
+      "caches": null,
+      "cores": null,
+      "hugepages": [
+       {
+        "num_pages": 1,
+        "page_size": 2048
+       },
+       {
+        "num_pages": 1,
+        "page_size": 1048576
+       }
+      ],
+      "memory": 33604804608,
+      "node_id": 1
+     }
+    ]
+    `)
+	assert.JSONEq(t, string(expectedTopology), string(topologyJSON))
```

Comment:
If you cast `expectedTopology` to string, then it could be reasonable to declare it as a string and get rid of casting.